src/Bridge/Symfony/Routing/IriConverter.php

```diff
@@ -112,10 +112,21 @@ public function getItemFromIri(string $iri, array $context = [])
 
     /**
      * {@inheritdoc}
+     *
+     * @deprecated use `getIriFromItemWithResource` instead to be removed in ApiPlatform 3.0
      */
     public function getIriFromItem($item, int $referenceType = UrlGeneratorInterface::ABS_PATH): string
     {
         $resourceClass = $this->getObjectClass($item);
+
+        return $this->getIriFromItemWithResource($item, $resourceClass, $referenceType);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getIriFromItemWithResource($item, string $resourceClass, int $referenceType = UrlGeneratorInterface::ABS_PATH): string
```

Comment:
WDYT about getIriFromItemWithContext? Where we'd add a $context['resource_class']? If we have other use cases in the future at least we won't have to change this interface again.

Reply:
I don't see any use case soon. I think it's confusing considering the main use case is to pass it from another context (serialization context). This seems good to me.

Also, I hate config-arrays. If you want it I'm in favor of creating a config object.

Reply:
Why not just keep the same interface but add an optional argument that's commented out?

Reply:
Why not, especially that it doesn't really make sense to **not** add the `resourceClass` to the `getIriFromItem`.